### PR TITLE
Added RPC support for Login Migration Validation API's

### PR DESCRIPTION
**Pull request by Bhavikshah123**
- Updated the new Login migration NuGet to 1.0.20240107.2 that exposes 4 public facing validation Api's
1. ValidateSysAdminPermissions
2.  ValidateAADDomainName
3. ValidateLoginEligibility
4. ValidateAADDomainName

- Created the RPC request for the validation API's 
1. ValidateSysAdminPermissionRequest
2. ValidateUserMappingRequest
3. ValidateAADDomainNameRequest
4. ValidateLoginEligibilityRequest

- Added the respective handlers for the validation API's
1. HandleSysAdminPermissionValidation
2. HandleAADDomainNameValidation
3. HandleLoginEligibilityValidation
4. HandleUserMappingValidation

- Added summaries to login migration RPC requests and handlers.

- Testing
Tested all the RPC calls to the validation API's by consuming the sqltoolsservice in the ADS client locally.

### CODE REVIEW

File: src/Microsoft.SqlTools.Migration/Contracts/StartLoginMigration.cs [hunk 1]
```diff
@@ -94,6 +119,24 @@ public class LoginMigrationResult
         public string ElapsedTime{ get; set; }
     }
 
+    public class LoginMigrationPreValidationResult
```
Comment (goyal-anjali) [inline src/Microsoft.SqlTools.Migration/Contracts/StartLoginMigration.cs:122]: Comment missing

File: src/Microsoft.SqlTools.Migration/Contracts/StartLoginMigration.cs [hunk 2]
```diff
@@ -108,6 +151,34 @@ public static readonly
                 RequestType<StartLoginMigrationParams, LoginMigrationResult>.Create("migration/validateloginmigration");
     }
 
+    public class ValidateSysAdminPermissionRequest
```
Comment (goyal-anjali) [inline src/Microsoft.SqlTools.Migration/Contracts/StartLoginMigration.cs:154]: Similar: Comment missing

File: src/Microsoft.SqlTools.Migration/MigrationService.cs [hunk 3]
```diff
@@ -320,6 +324,138 @@ public void InitializeService(IProtocolEndpoint serviceHost)
             }
         }
 
+        internal async Task HandleSysAdminPermissionValidation(
```
Comment (goyal-anjali) [inline src/Microsoft.SqlTools.Migration/MigrationService.cs:327]: Similar: Comment missing

**Comment by goyal-anjali** [comment]
Can we also add the testing part in the description?

File: src/Microsoft.SqlTools.Migration/MigrationService.cs [hunk 4]
```diff
@@ -320,6 +324,138 @@ public void InitializeService(IProtocolEndpoint serviceHost)
...
+                stopWatch.Start();
+                exceptionMap.AddExceptions(loginMigration.ValidateAADDomainName(parameters.AADDomainName));
+                stopWatch.Stop();
+                TimeSpan elapsedTime = stopWatch.Elapsed;
+
+                LoginMigrationPreValidationResult results = new LoginMigrationPreValidationResult()
+                {
+                    ExceptionMap = exceptionMap,
+                    CompletedStep = LoginMigrationPreValidationStep.AADDomainNameValidation,
+                    ElapsedTime = MigrationServiceHelper.FormatTimeSpan(elapsedTime)
```
[the remark this answers is not in the document]
Comment (Bhavikshah123) [reply src/Microsoft.SqlTools.Migration/MigrationService.cs:448]: Hours:minutes:seconds:milliseconds

File: src/Microsoft.SqlTools.Migration/MigrationService.cs [hunk 5]
```diff
@@ -320,6 +324,138 @@ public void InitializeService(IProtocolEndpoint serviceHost)
...
+            {
+                ILoginsMigrationLogger logger = this.GetLoginsMigrationLogger();
+                ILoginsMigration loginMigration = new LoginsMigration(parameters.SourceConnectionString, parameters.TargetConnectionString,
+                null, parameters.LoginList, parameters.AADDomainName, logger);
+
+                IDictionary<string, IEnumerable<LoginMigrationException>> exceptionMap = new Dictionary<string, IEnumerable<LoginMigrationException>>();
+
+                Stopwatch stopWatch = new Stopwatch();
+                stopWatch.Start();
+                exceptionMap.AddExceptions(loginMigration.ValidateUserMapping(CancellationToken.None));
```
Comment (ashutoshsuman99) [inline src/Microsoft.SqlTools.Migration/MigrationService.cs:374]: Do we follow any format for keeping all the validations as sync or all as async? As here I can see 2 are async and other 2 are sync.

Comment (Bhavikshah123) [reply src/Microsoft.SqlTools.Migration/MigrationService.cs:374]: There is no specific format. It varies based on the backend implementation.

**Review by goyal-anjali — APPROVED** [review]

**Pull request merged**